tensordict/nn/probabilistic.py

```diff
@@ -258,25 +261,28 @@ class ProbabilisticTensorDictModule(TensorDictModuleBase):
 
     """
 
-    SAMPLE_LOG_PROB_KEY = "sample_log_prob"
-
     def __init__(
         self,
-        in_keys: str | Sequence[str] | dict,
-        out_keys: str | Sequence[str] | None = None,
+        in_keys: NestedKey | Sequence[NestedKey] | dict,
+        out_keys: NestedKey | Sequence[NestedKey] | None = None,
         *,
         default_interaction_mode: str | None = None,
         default_interaction_type: InteractionType = InteractionType.MODE,
         distribution_class: type = Delta,
         distribution_kwargs: dict | None = None,
         return_log_prob: bool = False,
+        log_prob_key: Optional[NestedKey] = "sample_log_prob",
         cache_dist: bool = False,
         n_empirical_estimate: int = 1000,
     ) -> None:
         super().__init__()
-        if isinstance(in_keys, str):
+        if isinstance(in_keys, str) or (
+            isinstance(in_keys, tuple) and set(map(type, in_keys)) == {str}
+        ):
             in_keys = [in_keys]
```

Comment:
What are we trying to cover here? Why isn't a tuple check enough? Do we expect tuples with other things than strings?

Reply:
since we allow iterables the tuple could be the sequence of keys or a nested key itself. here we check it is the second case

Reply:
how can we differentiate 
```
{"a"}
```
from
```
{("a",)}
```
from
```
(("a",))
```
from
```
(["a",])
```
?

I think we should allow either: str/tuple => consider as single key // list => consider as list of keys

Reply:
ok yeah let's do this so we can "cut the tail to the cat".

Should I update the docstring to list of nested key?